cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.proto.security.Privilege;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Default implementation of {@link AuthorizationEnforcementService}.
+ */
+@Singleton
+public class DefaultAuthorizationEnforcementService extends AbstractScheduledService
+  implements AuthorizationEnforcementService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultAuthorizationEnforcementService.class);
+  private static final EnumSet<State> SERVICE_AVAILABLE_STATES =
+    EnumSet.of(State.STARTING, State.RUNNING, State.STOPPING);
+
+  private final AuthorizerInstantiator authorizerInstantiator;
+  private final boolean authorizationEnabled;
+  private final boolean cacheEnabled;
+  private final int cacheRefreshIntervalSecs;
+  private final LoadingCache<Principal, Set<Privilege>> authPolicyCache;
+
+  @Inject
+  DefaultAuthorizationEnforcementService(final AuthorizerInstantiator authorizerInstantiator,
+                                         CConfiguration cConf) {
+    this.authorizerInstantiator = authorizerInstantiator;
+    this.authorizationEnabled = cConf.getBoolean(Constants.Security.Authorization.ENABLED);
+    this.cacheEnabled = cConf.getBoolean(Constants.Security.Authorization.CACHE_ENABLED);
+    this.cacheRefreshIntervalSecs = cConf.getInt(Constants.Security.Authorization.CACHE_REFRESH_INTERVAL_SECS);
```

Comment:
You might want to validate that the interval and ttl are > 0. Also log a warning if ttl <= interval.